analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/NonNullableValueTypeCheck.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
+{
+    internal class NonNullableValueTypeCheck : SymbolicCheck
+    {
+        protected override ProgramState PreProcessSimple(SymbolicContext context)
+        {
+            var operation = context.Operation.Instance;
+            if (IPropertyReferenceOperationWrapper.IsInstance(operation)
```

Comment:
Needs to be extended to other reference kinds as well. Maybe "locals" can be omitted in the long run but everything else including parameter references should benefit from this approach.